### ISSUE TITLE
results

### ISSUE THREAD

**Issue by szh404** [Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation#5]
Hey, @Coordi777  
Regarding the test results, I have some doubts and hope you can provide some valuable suggestions.

I am using the SAR2Opt dataset, which consists of 1450 pairs of training samples.

![1_0_0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/fa89fdef-3156-47a2-aa82-b965f21102cb)   
![1_0_0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/62c94bbc-0936-4bc3-ac22-47343346ecc2)

I trained using the default parameters in the code you provided(Thank you).

However, when I sample, I find that the quality of the generated images is not good. Additionally, the values of SSIM and FID metrics are also poor.

model050000.pt
![0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/5c65b7d1-af34-413c-a254-0777462d988a)
![1](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/ea898163-eebf-405f-837a-4fc2e46f7d07)

model070000.pt
test_0.png
![0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/67285a7b-2fb4-4ab7-8981-c85ed24a77fd)
![0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/5fd4d957-0846-42b6-918d-94de124660a6)
![0](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/effad331-553b-45d1-bcac-5644ddde0ba0)

test_7.png
![7](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/d0380b19-3e42-4893-8ea9-22f153d73b7d)
![7](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/98b99021-4420-4367-839d-d4e80723ef86)
![7](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/96819433/57c279b4-8a74-4d5f-af7d-60f20a1793ec)

Is it because my training data amount is too small, leading to this phenomenon, or are there other settings reasons? I hope you can provide some advice.

**Comment by ZhangJinian** [comment]
Hi! May I ask what system you are running the code on? I tried running it on Windows but ran into difficulties.

**Comment by szh404** [comment]
> Hi! May I ask what system you are running the code on? I tried running it on Windows but ran into difficulties.

Ubuntu20.04， Pytorch1.11， Cuda11.3

**Comment by ZhangJinian** [comment]
Thank you！

> > Hi! May I ask what system you are running the code on? I tried running it on Windows but ran into difficulties.
> 
> Ubuntu20.04， Pytorch1.11， Cuda11.3

Thank you！

**Comment by ZhangJinian** [comment]
My training dataset has 32520pairs, and I have trained the model for 32950(model032950.pt.) My results are very bad. I think it’s because we don’t train it enough.

![1711702862860](https://github.com/Coordi777/Conditional-Diffusion-for-SAR-to-Optical-Image-Translation/assets/131846756/f9bfc89f-d72b-48db-89a4-e3b0e2d0c6f7)

**Comment by szh404** [comment]
> My training dataset has 32520pairs, and I have trained the model for 32950(model032950.pt.) My results are very bad. I think it’s because we don’t train it enough.
> 
> ![1711702862860](https://private-user-images.githubusercontent.com/131846756/318004845-f9bfc89f-d72b-48db-89a4-e3b0e2d0c6f7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzkyMjgsIm5iZiI6MTcxMTkzODkyOCwicGF0aCI6Ii8xMzE4NDY3NTYvMzE4MDA0ODQ1LWY5YmZjODlmLWQ3MmItNDhkYi04OWE0LWUzYjBlMmQwYzZmNy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMjM1MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZjZiYTUwMDY2NGNkYzc5NzIzZmQ0YmY3Nzc5MDIyZTZlMTY0M2M0MjZjYzY0ZDEwOGJjZjg2NDdhZDViMWVlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.G5qss1ZewsGQ5qc0XKrXxXSstFaqCuPIfWaEYHdS_uQ)

Sorry for my late reply. I was busy with other things a few days ago and didn't reply promptly. How is the reproduction effect now? Is it the same as before?
Regarding my SAR2OPT dataset, I trained for 100,000 iterations, and the performance of the model saved at 070000.pt was the best. However, it still didn't achieve the expected performance.

Best wishes!

**Comment by 879087734** [comment]
Brother, I would like to ask if you have succeeded in running and how it has worked

**Comment by szh404** [comment]
> Brother, I would like to ask if you have succeeded in running and how it has worked

Indeed, I has succeeded running the program, but the results wasn't satisfied and I don't know how to train (set hyper parameters) to get a good result. I'll carry on doing this.

**Comment by 879087734** [comment]
My code doesn't work, can we add contact information, I want to ask you some questions, are you Chinese? My QQ is 3032904759

**Comment by Anurag-Mishra22** [comment]
@szh404  can you suggest how to get best output. Now are you satisfied with results. Plz guide.

**Comment by Yyang-sys** [comment]
你好，请问训练到什么程度停止呢，保存的模型是需要自己在该项目下创建个文件夹目录然后修改export OPENAI_LOGDIR=/dirs/to/log路径吗

**Comment by Mizar29** [comment]
你好，我也出现了类似的问题，请问你最后跑出来了吗？
